### PR TITLE
1619 renewal notifications

### PR DESCRIPTION
**Pull request by serge-gaia**
- Bulk licence renewal notifications for officers from a new link in the license dashboard.
- This includes the possibility to filter the licences by expiration date.
- Fixed also a wee bug in the email.

### CODE REVIEW

File: wildlifelicensing/apps/dashboard/static/wl/js/dash_tables.js [hunk 1]
```diff
@@ -135,6 +139,21 @@ define(
                             console.log("Error while loading licences data:", thrownError, textStatus, xhr.responseText, xhr.status);
                             //Stop the data table 'Processing'.
                             $(options.selectors.licencesTable + '_processing').hide();
+                        },
+                        complete: function () {
+                            // trick. Set the url for the bulk renewal. The bulk renewal view works uses the same
+                            // parameters than a datatable
```
Comment (tony-gaia) [inline wildlifelicensing/apps/dashboard/static/wl/js/dash_tables.js]: Poor grammer on this comment.

Comment (serge-gaia) [inline wildlifelicensing/apps/dashboard/static/wl/js/dash_tables.js]: Poor **grammar** on this comment! :-)

File: wildlifelicensing/apps/dashboard/views/officer.py [hunk 2]
```diff
@@ -443,6 +458,26 @@ def filter_licence_type(value):
             return None
 
     @staticmethod
+    def filter_expiry_after(value):
+        if value:
+            try:
+                date = date_parse(value, dayfirst=True).date()
+                return Q(end_date__gt=date)
```
Comment (tony-gaia) [inline wildlifelicensing/apps/dashboard/views/officer.py]: should probably use gte to include the end date.

File: wildlifelicensing/apps/dashboard/views/officer.py [hunk 3]
```diff
@@ -443,6 +458,26 @@ def filter_licence_type(value):
...
+            except:
+                pass
+        return None
+
+    @staticmethod
+    def filter_expiry_before(value):
+        if value:
+            try:
+                date = date_parse(value, dayfirst=True).date()
+                return Q(end_date__lt=date)
```
Comment (tony-gaia) [inline wildlifelicensing/apps/dashboard/views/officer.py]: should probably use lte to include the end date.

File: wildlifelicensing/apps/dashboard/static/wl/js/dash_tables.js (hunk 1 above)
Comment (serge-gaia) [inline wildlifelicensing/apps/dashboard/static/wl/js/dash_tables.js]: Updated

File: wildlifelicensing/apps/dashboard/views/officer.py (hunk 3 above)
Comment (serge-gaia) [inline wildlifelicensing/apps/dashboard/views/officer.py]: done

**Pull request merged**